std/experimental/logger/asynclogger.d

```
@@ -0,0 +1,127 @@
+    void logFileDefault()
+    {
+        fl.log("test message");
+    }
+
+    auto afl = new AsyncLogger(new FileLogger("asyncFileLogger.log", LogLevel.all));
+    void logFileAsync()
+    {
+        afl.log("test message");
+    }
+
+    auto r = benchmark!(logFileDefault, logFileAsync)(100_000);
+    writeln(map!(a => to!string(to!Duration(a)))(r[]));
+}
+-------------
+*/
+final class AsyncLogger : Logger
+{
+    /// Shuntdown message
+    struct ShutdownMsg {}
+
+    private static void spawnedFunc(shared Logger ptr) @trusted
+    {
+        auto l = cast(Logger)(ptr);
+        bool isRunning = true;
+        while (isRunning)
+        {
+            receive(
+                (LogEntryBase e)
+                {
+                    LogEntry tmp = e;
+                    tmp.threadId = ownerTid;
+                    tmp.logger = l;
+                    l.forwardMsg(tmp);
+                },
+                (ShutdownMsg msg) { isRunning = false; },
+                (OwnerTerminated ex) { isRunning = false; },
+                (Variant v) { throw new Error("unhandled message"); }
+            );
+        }
+    }
+
+    /** This constructor sets a LogLevel and starts a new logging thread
+    that logs every entry asynchronously by using Logger. Because logger writes down
+    all entries into special logging thread, logger should not be used further
+    (without confidence that logger is not being used at the moment).
+    */
+    this(Logger logger, LogLevel lv = LogLevel.all) @trusted
+    {
+        super(lv);
+        this.logger_ = cast(shared Logger)(logger);
+        this.tid_ = spawn(&spawnedFunc, this.logger_);
+    }
+
+    override void writeLogMsg(ref LogEntry payload) @trusted
+    {
+        LogEntryBase msg = payload;
+        send(this.tid_, msg);
+    }
+
+    /// Stops logging thread by sending ShutdownMsg to it.
+    ~this() @trusted
+    {
+        send(this.tid_, ShutdownMsg());
```

having a try catch feels wrong. That will hide the error from the user. And what is a sane impl of the error handler? IMO, we need a test checking the currently expected behavior of sending to a dead Tid. If it changes the PR that introduces that change need to be checked.